lib/jxl/dec_reconstruct.cc

```diff
@@ -68,9 +68,17 @@ Status UndoXYBInPlace(Image3F* idct, const Rect& rect,
       }
     } else if (output_encoding_info.color_encoding.tf.IsSRGB()) {
       for (size_t x = 0; x < rect.xsize(); x += Lanes(d)) {
+#if MEMORY_SANITIZER
+        const auto mask = Iota(d, x) < Set(d, rect.xsize());
+        const auto sentinel = Set(d, kSanitizerSentinel);
+        const auto in_opsin_x = IfThenElse(mask, Load(d, row0 + x), sentinel);
+        const auto in_opsin_y = IfThenElse(mask, Load(d, row1 + x), sentinel);
+        const auto in_opsin_b = IfThenElse(mask, Load(d, row2 + x), sentinel);
+#else
         const auto in_opsin_x = Load(d, row0 + x);
         const auto in_opsin_y = Load(d, row1 + x);
         const auto in_opsin_b = Load(d, row2 + x);
+#endif
```

Comment:
Can we poison the portion of row{0,1,2} + x in line 91?

Reply:
Likely, yes. Do you propose to unpoison-process-poison here?

Reply:
yes. The unpoison-process-poison here will be founded on the fact that XybToRgb uses the undefined values but it is safe to do so; but we shouldn't assume that the output xybtorgb is initialized in other places because of this.
A comment about why next to the #ifdef MEMORY_SANITIZER explaining why would be appreciated.